File: typescript/nth-prime/nth-prime.ts
const isPrime = (n: number) => {
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i == 0) {
      return false;
    }
  }
  return true;
}

class Prime {
  nth(n: number) {
    if (n <= 0) {
      throw new Error('Prime is not possible');
    }
    const primes: number[] = [];
    let testValue: number = 2;
    while (primes.length < n) {
      if (isPrime(testValue++)) {
        primes.push(testValue - 1);
      }
    }
    return primes[n - 1];
  }
}

export default Prime

File: python/nth-prime/nth_prime.py
from math import sqrt

def is_prime(n: int) -> bool:
    for i in range(2, int(sqrt(n)) + 1):
        if n % i == 0:
            return False
    
    return True

def prime(number: int) -> int:
    if number < 1:
        raise ValueError("Error: Number must be positive")

    primes = [2]
    test_val = 3

    while len(primes) <= number:
        if is_prime(test_val):
            primes.append(test_val)
        test_val += 1

    return primes[number - 1]
